File: scripts/api_smoke_tests.py
#!/usr/bin/env python3
"""End-to-end API smoke checks for local Docker and deployed environments."""

from __future__ import annotations

import json
import os
import sys
import time
import urllib.error
import urllib.parse
import urllib.request


BASE_URL = os.environ.get("CFF_API_BASE_URL", "http://127.0.0.1:8080").strip().rstrip("/")
PASSWORD = os.environ.get("CFF_SMOKE_PASSWORD", "SmokeTest123!")
EMAIL_PREFIX = os.environ.get("CFF_SMOKE_EMAIL_PREFIX", "smoke")
ADMIN_API_TOKEN = os.environ.get("CFF_ADMIN_API_TOKEN", "").strip()


class SmokeFailure(RuntimeError):
    pass


def request(method: str, path: str, body=None, token: str = "", expected=(200,), extra_headers=None):
    data = None
    headers = {"Accept": "application/json"}
    if extra_headers:
        headers.update(extra_headers)
    if body is not None:
        data = json.dumps(body).encode("utf-8")
        headers["Content-Type"] = "application/json"
    if token:
        headers["Authorization"] = f"Bearer {token}"

    req = urllib.request.Request(
        f"{BASE_URL}{path}",
        data=data,
        headers=headers,
        method=method,
    )
    try:
        with urllib.request.urlopen(req, timeout=30) as response:
            text = response.read().decode("utf-8")
            status = response.getcode()
    except urllib.error.HTTPError as exc:
        text = exc.read().decode("utf-8")
        status = exc.code
    except urllib.error.URLError as exc:
        raise SmokeFailure(f"{method} {path} could not connect to {BASE_URL}: {exc}") from exc

    if status not in expected:
        raise SmokeFailure(f"{method} {path} expected {expected}, got {status}: {text}")
    if not text:
        return {}
    try:
        return json.loads(text)
    except json.JSONDecodeError:
        return {"raw": text}


def assert_true(condition, message: str):
    if not condition:
        raise SmokeFailure(message)


def smoke_player(suffix: str, player_id: str, name: str, position: str, projection: float = 10):
    return {
        "id": f"{player_id}-{suffix}",
        "name": name,
        "team": "Test State",
        "position": position,
        "conference": "Smoke",
        "projection": projection,
        "rank": 99,
    }


def operation_key(suffix: str, name: str) -> str:
    return f"smoke-{suffix}-{name}"


def main():
    suffix = str(int(time.time()))
    email = f"{EMAIL_PREFIX}+{suffix}@example.com"
    manager_email = f"{EMAIL_PREFIX}-manager+{suffix}@example.com"

    health = request("GET", "/health")
    assert_true(health.get("status") in {"ok", "degraded"}, f"unexpected health payload: {health}")

    api_health = request("GET", "/api/health")
    assert_true(api_health.get("service") == "college-ff-api", f"unexpected API health payload: {api_health}")
    request("GET", "/api/leagues", expected=(401,))

    signup = request("POST", "/api/auth/signup", {"email": email, "password": PASSWORD}, expected=(201,))
    token = signup.get("token")
    assert_true(token, f"signup did not return token: {signup}")

    validate = request("GET", "/api/auth/validate", token=token)
    assert_true(validate.get("valid") is True, f"token did not validate: {validate}")

    login = request("POST", "/api/auth/login", {"email": email, "password": PASSWORD})
    login_token = login.get("token")
    assert_true(login_token, f"login did not return token: {login}")
    request("POST", "/api/auth/logout", {}, token=login_token)
    request("GET", "/api/auth/validate", token=login_token, expected=(401,))

    league_payload = {
        "name": f"Smoke League {suffix}",
        "teams": 10,
        "scoring": "ppr",
        "draftType": "snake",
        "invitedEmails": [manager_email],
        "rosterRules": {"qb": 0, "rb": 0, "wr": 0, "te": 0, "flex": 0, "bench": 8},
        "waiverRules": {
            "mode": "waivers",
            "claimDeadline": "2000-01-01T00:00:00Z",
            "freeAgencyLocked": True,
        },
        "tradeRules": {
            "commissionerApproval": False,
            "expirationHours": 48,
        },
    }
    league = request("POST", "/api/leagues", league_payload, token=token, expected=(201,))
    league_id = league.get("id")
    assert_true(league_id, f"league creation did not return an ID: {league}")

    leagues = request("GET", "/api/leagues", token=token)
    assert_true(any(item.get("id") == league_id for item in leagues), "created league is missing from list")

    settings_update = {**league, "notes": "smoke settings update"}
    updated = request("PUT", f"/api/leagues/{league_id}", settings_update, token=token)
    assert_true(updated.get("notes") == "smoke settings update", f"league update failed: {updated}")

    manager_signup = request("POST", "/api/auth/signup", {"email": manager_email, "password": PASSWORD}, expected=(201,))
    manager_token = manager_signup.get("token")
    assert_true(manager_token, f"manager signup did not return token: {manager_signup}")

    joined = request("POST", f"/api/leagues/{league_id}/join", {}, token=manager_token, expected=(200, 202))
    assert_true(joined.get("id") == league_id, f"manager could not request league access: {joined}")
    if joined.get("joinStatus") == "pending_approval":
        approved = request(
            "PUT",
            f"/api/leagues/{league_id}/members/{urllib.parse.quote(manager_email, safe='')}",
            {"status": "active", "role": "member"},
            token=token,
        )
        assert_true(
            any(member.get("email") == manager_email and str(member.get("status", "")).lower() == "active" for member in approved),
            f"manager approval failed: {approved}",
        )
    else:
        assert_true(
            joined.get("joinStatus") == "active" or joined.get("id") == league_id,
            f"manager could not join invited league: {joined}",
        )

    members = request("GET", f"/api/leagues/{league_id}/members", token=token)
    member_emails = {member.get("email") for member in members}
    assert_true({email, manager_email}.issubset(member_emails), f"joined members missing: {members}")

    lobby_league = request("PUT", f"/api/leagues/{league_id}", {**updated, "draftLobbyOpen": True}, token=token)
    assert_true(lobby_league.get("draftLobbyOpen") is True, f"draft lobby did not open: {lobby_league}")

    draft_state = request("GET", f"/api/leagues/{league_id}/draft", token=token)
    order_state = request(
        "PUT",
        f"/api/leagues/{league_id}/draft/order",
        {"draftOrder": [email, manager_email], "expectedVersion": draft_state.get("version") or 0},
        token=token,
    )
    assert_true(order_state.get("draftOrder") == [email, manager_email], f"draft order was not saved: {order_state}")
    request("POST", f"/api/leagues/{league_id}/draft/readiness", {"ready": True}, token=token)
    draft_state = request("POST", f"/api/leagues/{league_id}/draft/readiness", {"ready": True}, token=manager_token)
    draft_state = request(
        "POST",
        f"/api/leagues/{league_id}/draft/start",
        {"expectedVersion": draft_state.get("version") or 0},
        token=token,
    )
    assert_true(draft_state.get("status") == "open", f"draft did not start: {draft_state}")

    commissioner_player = smoke_player(suffix, "smoke-rb", "Smoke Test RB", "RB", 18.4)
    manager_player = smoke_player(suffix, "smoke-wr", "Smoke Test WR", "WR", 17.2)
    extra_player = smoke_player(suffix, "smoke-qb", "Smoke Test QB", "QB", 22.1)

    first_pick = request(
        "POST",
        f"/api/leagues/{league_id}/draft/picks",
        {
            "player": commissioner_player,
            "expectedPick": draft_state.get("currentPick"),
            "expectedVersion": draft_state.get("version") or 0,
        },
        token=token,
        expected=(201,),
    )
    assert_true(first_pick.get("currentManager") == manager_email, f"pick 2 should belong to manager: {first_pick}")

    second_pick = request(
        "POST",
        f"/api/leagues/{league_id}/draft/picks",
        {
            "player": manager_player,
            "expectedPick": first_pick.get("currentPick"),
            "expectedVersion": first_pick.get("version") or 0,
        },
        token=manager_token,
        expected=(201,),
    )
    request(
        "POST",
        f"/api/leagues/{league_id}/draft/picks",
        {
            "player": smoke_player(suffix, "bad-turn", "Smoke Bad Turn", "TE"),
            "expectedPick": second_pick.get("currentPick"),
            "expectedVersion": second_pick.get("version") or 0,
        },
        token=token,
        expected=(409,),
    )
    snake_turn = request(
        "POST",
        f"/api/leagues/{league_id}/draft/picks",
        {
            "player": extra_player,
            "expectedPick": second_pick.get("currentPick"),
            "expectedVersion": second_pick.get("version") or 0,
        },
        token=manager_token,
        expected=(201,),
    )
    assert_true(snake_turn.get("currentManager") == email, f"pick 4 should return to commissioner: {snake_turn}")

    undo_state = request(
        "POST",
        f"/api/leagues/{league_id}/draft/undo",
        {"expectedVersion": snake_turn.get("version") or 0},
        token=token,
    )
    assert_true(len(undo_state.get("picks") or []) == 2, f"draft undo did not remove last pick: {undo_state}")

    pending_waiver_player = smoke_player(suffix, "waiver-pending", "Smoke Pending Waiver", "QB", 9.8)
    pending_waiver = request(
        "POST",
        f"/api/leagues/{league_id}/waivers",
        {"addPlayer": pending_waiver_player, "dropPlayerId": commissioner_player["id"]},
        token=token,
        expected=(200, 201),
        extra_headers={"Idempotency-Key": operation_key(suffix, "waiver-create")},
    )
    assert_true(pending_waiver.get("id"), f"pending waiver missing id: {pending_waiver}")

    request("POST", f"/api/leagues/{league_id}/score/week/1", {"season": 2026}, token=token)
    finalized = request("POST", f"/api/leagues/{league_id}/score/week/1/finalize", {}, token=token)
    assert_true(
        any(str(matchup.get("status", "")).lower() == "final" for matchup in finalized),
        f"week was not finalized: {finalized}",
    )

    request(
        "POST",
        f"/api/leagues/{league_id}/roster",
        {"player": smoke_player(suffix, "locked-add", "Smoke Locked Add", "TE")},
        token=token,
        expected=(409,),
        extra_headers={"Idempotency-Key": operation_key(suffix, "locked-add")},
    )
    request(
        "POST",
        f"/api/leagues/{league_id}/roster/drop",
        {"playerId": commissioner_player["id"]},
        token=token,
        expected=(409,),
        extra_headers={"Idempotency-Key": operation_key(suffix, "locked-drop")},
    )
    request(
        "POST",
        f"/api/leagues/{league_id}/roster/{commissioner_player['id']}/slot",
        {"slot": "bench"},
        token=token,
        expected=(409,),
        extra_headers={"Idempotency-Key": operation_key(suffix, "locked-slot")},
    )
    request(
        "POST",
        f"/api/leagues/{league_id}/waivers",
        {"addPlayer": smoke_player(suffix, "locked-waiver", "Smoke Locked Waiver", "TE"), "dropPlayerId": commissioner_player["id"]},
        token=token,
        expected=(409,),
        extra_headers={"Idempotency-Key": operation_key(suffix, "locked-waiver")},
    )
    request(
        "POST",
        f"/api/leagues/{league_id}/waivers/{pending_waiver['id']}/process",
        {},
        token=token,
        expected=(409,),
        extra_headers={"Idempotency-Key": operation_key(suffix, "waiver-process")},
    )
    request(
        "POST",
        f"/api/leagues/{league_id}/waivers/{pending_waiver['id']}/status",
        {"status": "Cancelled"},
        token=token,
        expected=(200,),
        extra_headers={"Idempotency-Key": operation_key(suffix, "waiver-cancel")},
    )
    request(
        "POST",
        f"/api/leagues/{league_id}/trades",
        {
            "offerPlayer": commissioner_player,
            "requestPlayer": manager_player,
            "requestPlayerName": manager_player["name"],
            "targetManager": manager_email,
        },
        token=token,
        expected=(409,),
        extra_headers={"Idempotency-Key": operation_key(suffix, "locked-trade")},
    )

    request("GET", "/api/admin/ingest/cfbd/status", token=token, expected=(403,))
    if ADMIN_API_TOKEN:
        admin_status = request("GET", "/api/admin/ingest/cfbd/status", token=ADMIN_API_TOKEN)
        assert_true(isinstance(admin_status, dict), f"unexpected admin ingestion status: {admin_status}")

    transactions = request("GET", f"/api/leagues/{league_id}/transactions", token=token)
    assert_true(isinstance(transactions, list), f"transactions response is not a list: {transactions}")

    reset = request("POST", "/api/auth/request-password-reset", {"email": email})
    reset_token = reset.get("passwordResetToken")
    if reset_token:
        new_password = f"{PASSWORD}Reset"
        request("POST", "/api/auth/reset-password", {"token": reset_token, "password": new_password})
        request("GET", "/api/auth/validate", token=token, expected=(401,))
        relogin = request("POST", "/api/auth/login", {"email": email, "password": new_password})
        assert_true(relogin.get("token"), f"password reset login failed: {relogin}")

    print(json.dumps({
        "status": "ok",
        "baseUrl": BASE_URL,
        "email": email,
        "managerEmail": manager_email,
        "leagueId": league_id,
    }, indent=2))


if __name__ == "__main__":
    try:
        main()
    except SmokeFailure as exc:
        print(json.dumps({"status": "failed", "error": str(exc)}, indent=2), file=sys.stderr)
        raise SystemExit(1)

File: scripts/api_smoke_tests.mjs
#!/usr/bin/env node

const baseUrl = (process.env.CFF_API_BASE_URL || 'http://127.0.0.1:8080').replace(/\/$/, '');
const password = process.env.CFF_SMOKE_PASSWORD || 'SmokeTest123!';
const emailPrefix = process.env.CFF_SMOKE_EMAIL_PREFIX || 'smoke';

class SmokeFailure extends Error {}

async function request(method, path, body = undefined, token = '', expected = [200], extraHeaders = {}) {
  const headers = { Accept: 'application/json', ...extraHeaders };
  const init = { method, headers };
  if (body !== undefined) {
    headers['Content-Type'] = 'application/json';
    init.body = JSON.stringify(body);
  }
  if (token) headers.Authorization = `Bearer ${token}`;

  let response;
  let text;
  try {
    response = await fetch(`${baseUrl}${path}`, init);
    text = await response.text();
  } catch (error) {
    throw new SmokeFailure(`${method} ${path} could not connect to ${baseUrl}: ${error.message}`);
  }

  if (!expected.includes(response.status)) {
    throw new SmokeFailure(`${method} ${path} expected ${expected.join('/')}, got ${response.status}: ${text}`);
  }
  if (!text) return {};
  try {
    return JSON.parse(text);
  } catch {
    return { raw: text };
  }
}

function assertTrue(condition, message) {
  if (!condition) throw new SmokeFailure(message);
}

function smokePlayer(suffix, id, name, position, projection = 10) {
  return {
    id: `${id}-${suffix}`,
    name,
    team: 'Test State',
    position,
    conference: 'Smoke',
    projection,
    rank: 99
  };
}

function operationKey(suffix, name) {
  return `smoke-${suffix}-${name}`;
}

async function main() {
  const suffix = String(Math.floor(Date.now() / 1000));
  const email = `${emailPrefix}+${suffix}@example.com`;
  const managerEmail = `${emailPrefix}-manager+${suffix}@example.com`;

  const health = await request('GET', '/health');
  assertTrue(['ok', 'degraded'].includes(health.status), `unexpected health payload: ${JSON.stringify(health)}`);

  const apiHealth = await request('GET', '/api/health');
  assertTrue(apiHealth.service === 'college-ff-api', `unexpected api health payload: ${JSON.stringify(apiHealth)}`);

  await request('GET', '/api/leagues', undefined, '', [401]);

  const signup = await request('POST', '/api/auth/signup', { email, password }, '', [201]);
  const token = signup.token;
  assertTrue(token, `signup did not return token: ${JSON.stringify(signup)}`);

  const validate = await request('GET', '/api/auth/validate', undefined, token);
  assertTrue(validate.valid === true, `token did not validate: ${JSON.stringify(validate)}`);

  const login = await request('POST', '/api/auth/login', { email, password });
  const loginToken = login.token;
  assertTrue(loginToken, `login did not return token: ${JSON.stringify(login)}`);

  await request('POST', '/api/auth/logout', {}, loginToken);
  await request('GET', '/api/auth/validate', undefined, loginToken, [401]);

  const leaguePayload = {
    name: `Smoke League ${suffix}`,
    teams: 10,
    scoring: 'ppr',
    draftType: 'snake',
    invitedEmails: [managerEmail],
    rosterRules: { qb: 0, rb: 0, wr: 0, te: 0, flex: 0, bench: 8 },
    waiverRules: {
      mode: 'waivers',
      claimDeadline: '2000-01-01T00:00:00Z',
      freeAgencyLocked: true
    },
    tradeRules: {
      commissionerApproval: false,
      expirationHours: 48
    }
  };
  const league = await request('POST', '/api/leagues', leaguePayload, token, [201]);
  const leagueId = league.id;
  assertTrue(leagueId, `league create did not return id: ${JSON.stringify(league)}`);

  const leagues = await request('GET', '/api/leagues', undefined, token);
  assertTrue(leagues.some((item) => item.id === leagueId), 'created league missing from list');

  const settingsUpdate = { ...league, notes: 'smoke settings update' };
  const updated = await request('PUT', `/api/leagues/${leagueId}`, settingsUpdate, token);
  assertTrue(updated.notes === 'smoke settings update', `league update failed: ${JSON.stringify(updated)}`);

  const managerSignup = await request('POST', '/api/auth/signup', { email: managerEmail, password }, '', [201]);
  const managerToken = managerSignup.token;
  assertTrue(managerToken, `manager signup did not return token: ${JSON.stringify(managerSignup)}`);

  const joined = await request('POST', `/api/leagues/${leagueId}/join`, {}, managerToken, [200, 202]);
  assertTrue(joined.id === leagueId, `manager could not request league access: ${JSON.stringify(joined)}`);
  if (joined.joinStatus === 'pending_approval') {
    const approved = await request(
      'PUT',
      `/api/leagues/${leagueId}/members/${encodeURIComponent(managerEmail)}`,
      { status: 'active', role: 'member' },
      token
    );
    assertTrue(
      approved.some((member) => member.email === managerEmail && String(member.status).toLowerCase() === 'active'),
      `manager approval failed: ${JSON.stringify(approved)}`
    );
  } else {
    assertTrue(joined.joinStatus === 'active' || joined.id === leagueId, `manager could not join invited league: ${JSON.stringify(joined)}`);
  }

  const members = await request('GET', `/api/leagues/${leagueId}/members`, undefined, token);
  const memberEmails = new Set(members.map((member) => member.email));
  assertTrue(memberEmails.has(email) && memberEmails.has(managerEmail), `joined members missing: ${JSON.stringify(members)}`);

  const lobbyLeague = await request('PUT', `/api/leagues/${leagueId}`, { ...updated, draftLobbyOpen: true }, token);
  assertTrue(lobbyLeague.draftLobbyOpen === true, `draft lobby did not open: ${JSON.stringify(lobbyLeague)}`);

  let draftState = await request('GET', `/api/leagues/${leagueId}/draft`, undefined, token);
  const orderState = await request(
    'PUT',
    `/api/leagues/${leagueId}/draft/order`,
    { draftOrder: [email, managerEmail], expectedVersion: draftState.version || 0 },
    token
  );
  assertTrue(JSON.stringify(orderState.draftOrder) === JSON.stringify([email, managerEmail]), `draft order was not saved: ${JSON.stringify(orderState)}`);
  await request('POST', `/api/leagues/${leagueId}/draft/readiness`, { ready: true }, token);
  draftState = await request('POST', `/api/leagues/${leagueId}/draft/readiness`, { ready: true }, managerToken);
  draftState = await request(
    'POST',
    `/api/leagues/${leagueId}/draft/start`,
    { expectedVersion: draftState.version || 0 },
    token
  );
  assertTrue(draftState.status === 'open', `draft did not start: ${JSON.stringify(draftState)}`);

  const commPlayer = smokePlayer(suffix, 'smoke-rb', 'Smoke Test RB', 'RB', 18.4);
  const managerPlayer = smokePlayer(suffix, 'smoke-wr', 'Smoke Test WR', 'WR', 17.2);
  const extraPlayer = smokePlayer(suffix, 'smoke-qb', 'Smoke Test QB', 'QB', 22.1);

  const firstPick = await request('POST', `/api/leagues/${leagueId}/draft/picks`, {
    player: commPlayer,
    expectedPick: draftState.currentPick,
    expectedVersion: draftState.version || 0
  }, token, [201]);
  assertTrue(firstPick.currentManager === managerEmail, `pick 2 should belong to manager: ${JSON.stringify(firstPick)}`);

  const secondPick = await request('POST', `/api/leagues/${leagueId}/draft/picks`, {
    player: managerPlayer,
    expectedPick: firstPick.currentPick,
    expectedVersion: firstPick.version || 0
  }, managerToken, [201]);
  await request('POST', `/api/leagues/${leagueId}/draft/picks`, {
    player: smokePlayer(suffix, 'bad-turn', 'Smoke Bad Turn', 'TE'),
    expectedPick: secondPick.currentPick,
    expectedVersion: secondPick.version || 0
  }, token, [409]);
  const snakeTurn = await request('POST', `/api/leagues/${leagueId}/draft/picks`, {
    player: extraPlayer,
    expectedPick: secondPick.currentPick,
    expectedVersion: secondPick.version || 0
  }, managerToken, [201]);
  assertTrue(snakeTurn.currentManager === email, `pick 4 should return to commissioner: ${JSON.stringify(snakeTurn)}`);

  const undoState = await request('POST', `/api/leagues/${leagueId}/draft/undo`, {
    expectedVersion: snakeTurn.version || 0
  }, token);
  assertTrue((undoState.picks || []).length === 2, `draft undo did not remove last pick: ${JSON.stringify(undoState)}`);

  const pendingWaiverPlayer = smokePlayer(suffix, 'waiver-pending', 'Smoke Pending Waiver', 'QB', 9.8);
  const pendingWaiver = await request('POST', `/api/leagues/${leagueId}/waivers`, {
    addPlayer: pendingWaiverPlayer,
    dropPlayerId: commPlayer.id
  }, token, [200, 201], { 'Idempotency-Key': operationKey(suffix, 'waiver-create') });
  assertTrue(pendingWaiver.id, `pending waiver missing id: ${JSON.stringify(pendingWaiver)}`);

  await request('POST', `/api/leagues/${leagueId}/score/week/1`, { season: 2026 }, token);
  const finalized = await request('POST', `/api/leagues/${leagueId}/score/week/1/finalize`, {}, token);
  assertTrue(finalized.some((matchup) => String(matchup.status).toLowerCase() === 'final'), `week was not finalized: ${JSON.stringify(finalized)}`);

  await request('POST', `/api/leagues/${leagueId}/roster`, { player: smokePlayer(suffix, 'locked-add', 'Smoke Locked Add', 'TE') }, token, [409], { 'Idempotency-Key': operationKey(suffix, 'locked-add') });
  await request('POST', `/api/leagues/${leagueId}/roster/drop`, { playerId: commPlayer.id }, token, [409], { 'Idempotency-Key': operationKey(suffix, 'locked-drop') });
  await request('POST', `/api/leagues/${leagueId}/roster/${commPlayer.id}/slot`, { slot: 'bench' }, token, [409], { 'Idempotency-Key': operationKey(suffix, 'locked-slot') });
  await request('POST', `/api/leagues/${leagueId}/waivers`, { addPlayer: smokePlayer(suffix, 'locked-waiver', 'Smoke Locked Waiver', 'TE'), dropPlayerId: commPlayer.id }, token, [409], { 'Idempotency-Key': operationKey(suffix, 'locked-waiver') });
  await request('POST', `/api/leagues/${leagueId}/waivers/${pendingWaiver.id}/process`, {}, token, [409], { 'Idempotency-Key': operationKey(suffix, 'waiver-process') });
  await request('POST', `/api/leagues/${leagueId}/waivers/${pendingWaiver.id}/status`, { status: 'Cancelled' }, token, [200], { 'Idempotency-Key': operationKey(suffix, 'waiver-cancel') });
  await request('POST', `/api/leagues/${leagueId}/trades`, {
    offerPlayer: commPlayer,
    requestPlayer: managerPlayer,
    requestPlayerName: managerPlayer.name,
    targetManager: managerEmail
  }, token, [409], { 'Idempotency-Key': operationKey(suffix, 'locked-trade') });

  await request('GET', '/api/admin/ingest/cfbd/status', undefined, token, [403]);

  const transactions = await request('GET', `/api/leagues/${leagueId}/transactions`, undefined, token);
  assertTrue(Array.isArray(transactions), `transactions response is not a list: ${JSON.stringify(transactions)}`);

  const reset = await request('POST', '/api/auth/request-password-reset', { email });
  if (reset.passwordResetToken) {
    const newPassword = `${password}Reset`;
    await request('POST', '/api/auth/reset-password', { token: reset.passwordResetToken, password: newPassword });
    await request('GET', '/api/auth/validate', undefined, token, [401]);
    const relogin = await request('POST', '/api/auth/login', { email, password: newPassword });
    assertTrue(relogin.token, `password reset login failed: ${JSON.stringify(relogin)}`);
  }

  console.log(JSON.stringify({
    status: 'ok',
    baseUrl,
    email,
    managerEmail,
    leagueId
  }, null, 2));
}

main().catch((error) => {
  const payload = {
    status: 'failed',
    error: error.message
  };
  console.error(JSON.stringify(payload, null, 2));
  process.exit(1);
});
